### PR TITLE
Upgrade to Java 17

### PR DESCRIPTION
**Pull request by JoergSiebahn**
As Sonar is executed from master, it runs with Java 11 and fails. That's why we don't have a Sonar report here and the check is missing. Local SonarLint IntelliJ plugin has no issues.

### CODE REVIEW

**Review by sliesche — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -105,7 +105,7 @@ test {
 wrapper { gradleVersion = '6.5' }
```
Comment (christopher-cudennec) [inline build.gradle:105]: I don't think it's in-sync with the actual Gradle wrapper version. I'm also not a fan of defining the wrapper in the `build.gradle`. As the `wrapper update` will not change this value, I think it's misleading. I'd vote for removing it. Or do you have any good reasons to keep it?

File: src/main/java/com/sdase/k8s/operator/mongodb/controller/tasks/CreateDatabaseTask.java [hunk 2]
```diff
@@ -2,44 +2,9 @@
...
-
-  public String getPassword() {
-    return password;
-  }
-
-  public String getConnectionString() {
-    return connectionString;
-  }
-}
+public record CreateDatabaseTask(
```
Comment (christopher-cudennec) [inline src/main/java/com/sdase/k8s/operator/mongodb/controller/tasks/CreateDatabaseTask.java:5]: Oh, Java 17 goodies 😄

File: build.gradle (hunk 1 above)
Comment (JoergSiebahn) [reply build.gradle:105]: No, I did not see it.

File: build.gradle [hunk 3]
```diff
@@ -17,15 +17,15 @@ idea.module.inheritOutputDirs = true
 
 version = System.getenv('SEMANTIC_VERSION')
 
-sourceCompatibility = JavaVersion.VERSION_11
+sourceCompatibility = JavaVersion.VERSION_17
```
[the remark this answers is not in the document]
Comment (JoergSiebahn) [reply build.gradle]: Updated

File: build.gradle (hunk 1 above)
Comment (JoergSiebahn) [reply build.gradle]: removed

File: src/main/java/com/sdase/k8s/operator/mongodb/controller/tasks/CreateDatabaseTask.java (hunk 2 above)
Comment (JoergSiebahn) [reply src/main/java/com/sdase/k8s/operator/mongodb/controller/tasks/CreateDatabaseTask.java:5]: And all done by IntelliJ shortcuts 🤟

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**